test/Microsoft.ML.Tests/Scenarios/IrisPlantClassificationTests.cs

```diff
@@ -65,53 +77,20 @@ public void TrainAndPredictIrisModelTest()
             Assert.Equal(.8, prediction.PredictedLabels[1], 1);
             Assert.Equal(0, prediction.PredictedLabels[2], 2);
 
-            // Note: Testing against the same data set as a simple way to test evaluation.
-            // This isn't appropriate in real-world scenarios.
-            string testDataPath = GetDataPath("iris.txt");
-            var testData = new TextLoader(testDataPath).CreateFrom<IrisData>(useHeader: false);
-
-            var evaluator = new ClassificationEvaluator();
-            evaluator.OutputTopKAcc = 3;
-            ClassificationMetrics metrics = evaluator.Evaluate(model, testData);
+            // Evaluate the trained pipeline
+            var predicted = trainedModel.Transform(testData);
+            var metrics = mlContext.MulticlassClassification.Evaluate(predicted, topK:3);
 
             Assert.Equal(.98, metrics.AccuracyMacro);
             Assert.Equal(.98, metrics.AccuracyMicro, 2);
-            Assert.Equal(.06, metrics.LogLoss, 2);
+            Assert.InRange(metrics.LogLoss, .05, .06);
             Assert.InRange(metrics.LogLossReduction, 94, 96);
-            Assert.Equal(1, metrics.TopKAccuracy);
 
             Assert.Equal(3, metrics.PerClassLogLoss.Length);
             Assert.Equal(0, metrics.PerClassLogLoss[0], 1);
             Assert.Equal(.1, metrics.PerClassLogLoss[1], 1);
             Assert.Equal(.1, metrics.PerClassLogLoss[2], 1);
-
-            ConfusionMatrix matrix = metrics.ConfusionMatrix;
-            Assert.Equal(3, matrix.Order);
-            Assert.Equal(3, matrix.ClassNames.Count);
-            Assert.Equal("0", matrix.ClassNames[0]);
-            Assert.Equal("1", matrix.ClassNames[1]);
-            Assert.Equal("2", matrix.ClassNames[2]);
-
-            Assert.Equal(50, matrix[0, 0]);
-            Assert.Equal(50, matrix["0", "0"]);
-            Assert.Equal(0, matrix[0, 1]);
-            Assert.Equal(0, matrix["0", "1"]);
-            Assert.Equal(0, matrix[0, 2]);
-            Assert.Equal(0, matrix["0", "2"]);
-
-            Assert.Equal(0, matrix[1, 0]);
-            Assert.Equal(0, matrix["1", "0"]);
-            Assert.Equal(48, matrix[1, 1]);
-            Assert.Equal(48, matrix["1", "1"]);
-            Assert.Equal(2, matrix[1, 2]);
-            Assert.Equal(2, matrix["1", "2"]);
-
-            Assert.Equal(0, matrix[2, 0]);
-            Assert.Equal(0, matrix["2", "0"]);
-            Assert.Equal(1, matrix[2, 1]);
-            Assert.Equal(1, matrix["2", "1"]);
-            Assert.Equal(49, matrix[2, 2]);
-            Assert.Equal(49, matrix["2", "2"]);
```

Comment:
Do we not have confusion matrix in metrics?

Reply:
I do not see it inside `metrics`.  Who would be the best person to confirm  whether we support `ConfusionMatrix` in the new API ?

If we do, is there an example I can follow ? I do not see it in the CookBook either

---
In reply to: [245114114](https://github.com/dotnet/machinelearning/pull/2008#discussion_r245114114) [](ancestors = 245114114)

Reply:
Created issue #2009  to bring back support for `ConfusionMatrix` in the new API 

---
In reply to: [245117612](https://github.com/dotnet/machinelearning/pull/2008#discussion_r245117612) [](ancestors = 245117612,245114114)